packages/jaeger-ui/src/utils/ant-design.tsx

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2020 The Jaeger Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { DefaultOptionType } from 'antd/lib/select';
+
+/**
+ * This enables search by label in Ant Design's Select component (v4.x)
+ *
+ * Pass this to the `filterOption` prop with `showSearch` prop set to `true`
+ *
+ * @example
+ * ```
+ * <Select {...props} filterOption={antSelectSearchByLabel} showSearch={true} />
```

Comment:
instead of providing this util function for search, could we instead write a function that decorates the Ant Select component in a way that it's always searchable, i.e. somewhat similar to `reduxFormFieldAdapter` function?

Reply:
Yes, that'd be better I guess, what should I name it? `CustomSelect` or `SearchableSelect` 

Also, if I create a separate wrapper, do I still keep the util file or just move everything to the wrapper?

Reply:
A very important principle (also related to the discussion in #1738) is this: never make functionality public unless absolutely necessary (aka "premature generalization"). You can always make something public later if needed, but the reverse is not true. So right now the only need we have is a new SearchableSelect component. The fact that it would use a filter function like this one is an implementation detail, which should be kept private to the module, until a strong use case appears that really needs it (highly doubtful considering we're talking about one liner).

Reply:
That makes quite a lot of sense, "premature generalization" - this is something new that I learned, thanks 🚀